compiler/src/dotty/tools/dotc/parsing/Scanners.scala

```diff
@@ -631,7 +631,11 @@ object Scanners {
       nextChar()
       if (ch == '/') { skipLine(); finishComment() }
       else if (ch == '*') { nextChar(); skipComment(); finishComment() }
-      else false
+      else {
+        // This was not a comment, remove the `/` from the buffer
+        commentBuf.clear()
```

Comment:
I am not sure you can clear the buffer. This method call itself recursively for nested comment. What about removing the last character or not adding it in the first place if it is not a comment?

Reply:
> This method call itself recursively for nested comment

That's what I thought too, and this is why I initially fixed it outside of `def skipComment()`. But it does not. There's an inner `skipComment`, but it will not be called unless we're actually inside a comment.

This function used to just return `false` to say "that's not a comment". Now it'll empty its buffer and say "that's not a comment". I think that should be safe, because the content of the buffer doesn't matter if we're not inside a comment, and it should actually be empty before we start reading the next comment.

Reply:
Oh yes right! This is confusing